src/XR/features/WebXRControllerPointerSelection.ts

```diff
@@ -366,14 +422,63 @@ export class WebXRControllerPointerSelection extends WebXRAbstractFeature {
                     scene.pointerY = this._screenCoordinatesRef.y;
                 }
             }
-            controllerData.pick = this._scene.pickWithRay(controllerData.tmpRay, this._scene.pointerMovePredicate || this.raySelectionPredicate);
 
-            const pick = controllerData.pick;
+            let pick = null;
+            let hoverAtOrigin = false;
+            let hoverRange = false; 
+            if (controllerData.hoverIndexMeshTip) {
+                let hoverPickInfo = this._pickWithMesh(controllerData.hoverIndexMeshTip, false, this.nearPickPredicate);
+                let hoverGrabInfo = this._pickWithMesh(controllerData.hoverIndexMeshTip, false, this.nearGrabPredicate);
+                if ((hoverPickInfo && hoverPickInfo.hit) || (hoverGrabInfo && hoverGrabInfo.hit)) {
+                    hoverRange = true;
+                } 
+                if (hoverRange) {
+                    controllerData.laserPointer.isVisible = false;
+                }
 
-            if (pick && pick.pickedPoint && pick.hit) {
-                // Update laser state
-                this._updatePointerDistance(controllerData.laserPointer, pick.distance);
+                let nearHover = hoverPickInfo && hoverPickInfo.pickedPoint && hoverPickInfo.hit;
+                if (hoverPickInfo?.pickedPoint) {
+                    hoverAtOrigin = hoverPickInfo.pickedPoint.x === 0 && hoverPickInfo.pickedPoint.y === 0 && hoverPickInfo.pickedPoint.z === 0;
+                }
+                if (nearHover && !hoverAtOrigin) {   
+                    controllerData.nearHover = true;
+                    pick = hoverPickInfo;
+                }
+            }
+
+            if (controllerData.pickIndexMeshTip) {
+                let pickInfo = this._pickWithMesh(controllerData.pickIndexMeshTip, false, this.nearPickPredicate);
+                let nearPick = pickInfo && pickInfo.pickedPoint && pickInfo.hit;
+                let pickAtOrigin = false;
+                if (pickInfo?.pickedPoint) {
+                    pickAtOrigin = pickInfo.pickedPoint.x === 0 && pickInfo.pickedPoint.y === 0 && pickInfo.pickedPoint.z === 0;
+                }
+                if (nearPick && !pickAtOrigin) {
+                    controllerData.nearPick = true;
+                    pick = pickInfo;
+                }
+            }
+
+            if (controllerData.grabRay) {
+                let pinchInfo = this._scene.pickWithRay(controllerData.grabRay, this.nearGrabPredicate);
+                if (pinchInfo && pinchInfo.pickedPoint && pinchInfo.hit) {  

```

Comment:
No check for a pick at the origin here?

Reply:
Also this section looks like it's close enough to the previous two to share the same lambda

Reply:
grab is using pickWithRay, so it does not issues with origin. the problem for pickwithmesh is that it uses the indextip position that is created at origin when initializing and it is causeing false positive. Pickwithray uses ray and it does not have to deal with this.

Reply:
Oh, I missed the fact that it uses a different picking method